### ISSUE TITLE
Fix KV-cache attention_mask off-by-one in multi-turn inference

### ISSUE THREAD

**Issue by sankpal-shreyas** [Shreyas582/WraithRun#114]
## Summary

Multi-turn live inference crashes on **every** model (GPT-2, TinyLlama 1.1B, Llama 3.2 1B) and **every** backend (CPU, DirectML) with:

```
BroadcastIterator::Init axis == 1 || axis == largest was false.
Attempting to broadcast an axis by a dimension other than 1. N by N+1
```

## Root Cause

In `inference_bridge/src/onnx_vitis.rs`, the decode loop sets:
- `attention_len = context_ids.len()` → should be `context_ids.len() + 1`
- `position_ids = [N-1]` → should be `[N]`

The attention mask has length N (past cache size) but the model expects N+1 (past + current token). This is an off-by-one that breaks the first cached-decode step on every run.

## Impact

**Live mode is completely non-functional** for any investigation requiring > 1 tool call (i.e., all real investigations). The ReAct agent loop always crashes after the first tool observation is appended to the transcript.

## Acceptance Criteria

- [ ] `attention_len` accounts for the current decode token: `context_ids.len() + 1`
- [ ] `position_ids` set to `[N]` not `[N-1]`
- [ ] Multi-turn investigation completes 3+ tool calls on CPU with GPT-2
- [ ] Multi-turn investigation completes on DirectML
- [ ] Integration test: 3-step ReAct loop with mock model
